cmd/soroban-cli/src/config/self_outdated_check.rs

```diff
@@ -0,0 +1,82 @@
+use crate::config::locator;
+use jsonrpsee_core::Serialize;
+use semver::Version;
+use serde::Deserialize;
+use std::fs;
+
+const FILE_NAME: &str = "self_outdated_check.toml";
+
+/// The `SelfOutdatedCheck` struct represents the state of the self-outdated check.
+/// This state is global and stored in the `self_outdated_check.toml` file in
+/// the global configuration directory.
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct SelfOutdatedCheck {
```

Comment:
Could we rename this type `UpdateCheck`? I think the name communicates the intent a little clearer.